### DISCUSSION TITLE
Table SelectFilter with deep relation

### DISCUSSION THREAD

**Discussion by cleargoal** [filamentphp/filament#15188] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

In my project, there are models with relations:

1. Contribution->belongsTo(Payment::class)
2. Payment->belongsTo(Operation::class)

I tried to set up a filter for the 'contributions' table in this way.
```
    SelectFilter::make('payment.operation_id')->label('Операція')
         ->relationship('payment.operation', 'title')
         ->options(Operation::query()->pluck('title', 'id')->all()),
```
The Filter is shown in his place and Operations are visible as options, but it doesn't filter.

**Please, show me my mistake(s).**

**Comment by leandrocfe** [comment]
Use a [filter form](https://filamentphp.com/docs/3.x/tables/filters/custom#custom-filter-forms) and customize the query to achieve what you want.

**Reply by cleargoal** [reply]
Thank you very much for your advice!
I tried to do it this way
```
    Filter::make('operation')->form([
        Select::make('operation_id')->label('Операція')
            ->options(Operation::query()->pluck('title', 'id')->all())
                ])->query(function (Builder $query, array $data): Builder {
                    return $query->where(fn (Builder $query, $payment): Builder => $query->where($payment->operation_id, 
                           $data['operation_id']), $data['id']);
                }),
```
but I see, that I don't understand, how to pass 'id' from options() to the query.
Can you help with this, please?
